x/ibc/07-tendermint/misbehaviour.go

```diff
@@ -78,19 +79,20 @@ func checkMisbehaviour(
 		return errors.New("unbonding period since last consensus state timestamp is over")
 	}
 
-	// Evidence is within the trusting period. ValidatorSet must have 2/3 similarity with trusted FromValidatorSet
-	// check that the validator sets on both headers are valid given the last trusted validatorset
-	// less than or equal to evidence height
-	if err := consensusState.ValidatorSet.VerifyFutureCommit(
-		evidence.Header1.ValidatorSet, evidence.ChainID,
-		evidence.Header1.Commit.BlockID, evidence.Header1.Height, evidence.Header1.Commit,
+	// TODO: - Evidence must be within trusting period
```

Comment:
This PR or another one?

Reply:
This PR

Reply:
Evidence must be within trusting period of the first header (`consensusState`)

Reply:
> Evidence must be within trusting period of the first header (consensusState)
This is fine

So we retrieve the last ConsensusState we stored before the evidence and retrieve its timestamp
However the Evidence itself contains 2 headers for the same height, with potentially different timestamps
what should be considered the timestamp of the evidence itself?
Should it be the earlier timestamp or later timestamp
Also, i don’t quite understand the point of using the timestamp here. Because this is already a malicious validator set we’re talking about. Can’t they arbitrarily make the timestamp fail whatever check on SubmitMisbehavior we have so that the client can’t be frozen?

Reply:
Sorry, I misunderstood your question at first. I think the issue here is as to what the limitation on submitting "would-have-been-fooled" evidence for old headers (potentially not the most recent) should be - for that it seems like we should require that the old header (~= consensus state) is no more than a `trustingPeriod` old (and necessarily that the two headers in evidence aren't from the future), since another client elsewhere could potentially have the old header and update to either of these new (equivocated) ones. It's also possible that the other client has a slightly older header but could update to another header in the middle and then to either of the forks though.

Reply:
Ref https://github.com/cosmos/ics/issues/379. For now let's just require that this `consensusState` be relatively recent.

Reply:
We still need to actually check this though (`consensusState.timestamp > ctx.BlockTime() - trustingPeriod`)

Reply:
Let's deal with this in a follow-up. Added a link to the ICS issue.